text/0138-achieving-order-between-pageload-and-srr-spans.md

```diff
@@ -0,0 +1,74 @@
+- Start Date: 2024-07-12
+- RFC Type: informational
+- RFC PR: https://github.com/getsentry/rfcs/pull/138
+- RFC Status: draft
+
+# Summary
+
+The hierarchy between Pageload Spans and SSR Spans is currently very confusing.
+SSR Spans are the ancestors of Pageload Spans, even though in a real-world scenario the Pageload is always the first thing that happens.
+This document discusses how this came to be and how it could be improved.
```

Comment:
This sounds like a tradeoff between correctness and what someone might intuitively expect. My initial thought seeing that screenshot was that pageload would be the parent span, but when you think about it, it makes perfect sense that it is not.

Reply:
In my opinion, it is not entirely just a tradeoff between correctness and intuition.

Just by looking at the order of execution, currently, the trace has the wrong topology: First the browser starts requesting the page, then the server responds. The trace shows it the other way around.
SDK-wise the trace is currently correct: The server SDK initializes before the browser SDK initializes.

The thing is, users likely don't care about the SDK, they care about the timings and topology of their application.

Reply:
>The thing is, users likely don't care about the SDK, they care about the timings and topology of their application.
Agree. Do we know how other providers tackle this problem, do they just ignore it?